### PR TITLE
feat(appsync): support for read consistency during DynamoDB reads

### LINKED ISSUE #5980: AppSync L2: support for read consistency
Allow specifying the `consistentRead` toggle in the `GetItem`, `ScanTable`, and `Query` mapping templates.

### Use Case

To control the amount of replicas contacted during DynamoDB reads.

### Proposed Solution

Add an enum parameter to the `dynamoDbGetItem`, `dynamoDbScanTable` and `dynamoDbQuery` methods.

Depends on #5861.

### Other

* [X] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by YichenQian09**
----
Motivation: DynamoDB uses eventually consistent reads unless you specify otherwise. Now the users cannot specify consistent read during DynamoDB reads. This commit allows users to set consistent read while creating resolvers.

Use case: To control the amount of replicas contacted during DynamoDB reads.

most salient design aspects: 
Added an boolean parameter to the dynamoDbGetItem, dynamoDbScanTable and dynamoDbQuery methods in mapping-template.ts. Changed relative tests.

closes #5980

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features
I ran yarn tests. They include integration test snapshots for DynamoDB reads. Do I need to write a new integration test?
* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by indrora** [comment]
Overall looks fine, but I'd rather see a test that shows that passing `consistentRead = true` in results in the change, not just changing all the calls to have `true` as their single argument. 

w/r/t the README, as a nit: It might be nice to have a link to the documentation ( https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference-dynamodb.html ) on how resolver mappings work and how you're passing them under the hood.

**Review by indrora — CHANGES_REQUESTED** [review]
I'd like to see at least in integ snapshots a test with none/true/false resolvers just for consistency across the full suite.

File: packages/@aws-cdk/aws-appsync/README.md [hunk 1]
```diff
@@ -94,8 +96,20 @@ demoDS.createResolver({
...
   responseMappingTemplate: appsync.MappingTemplate.dynamoDbResultItem(),
 });
+
+//To enable DynamoDB read consistency with the `MappingTemplate`:
+demoDS.createResolver({
+  typeName: 'Query',
+  fieldName: 'getDemosConsistent',
+  requestMappingTemplate: appsync.MappingTemplate.dynamoDbScanTable(true),
+  //requestMappingTemplate: appsync.MappingTemplate.dynamoDbGetItem('id', 'id', true),
+  //requestMappingTemplate: appsync.MappingTemplate.dynamoDbQuery(KeyCondition.eq('order', 'order'), 'orderIndex', true),
```
Comment (pahud) [inline packages/@aws-cdk/aws-appsync/README.md:106]: The two lines are a little bit duplicate as comments. Maybe just remove them?

```suggestion
```

**Review by pahud — CHANGES_REQUESTED** [review]
I am not sure if we should remove the comments in the README. I am fine with either.

**Review by indrora — APPROVED** [review]
LGTM. 

This covers every base and does a fantastic job of making sure it covers all the corner cases. 👍

**Pull request merged**